beps/bep_0007.rst

```diff
@@ -23,32 +24,21 @@ be interpreted as described in IETF `RFC 2119`_.
 
 .. _`RFC 2119`: http://tools.ietf.org/html/rfc2119
 
-Announce Parameter
-==================
-
-The client MAY add an ``&ipv6=`` parameter to the HTTP GET request it sends
-to the tracker. The value is either an IPv6 endpoint (address and port) or
-just an IPv6 address. In the case where only an address is supplied, the IPv6
-port is assumed to be the same as specified by the ``&port=`` parameter.
-
-The tracker SHOULD perform a NAT check on the IPv6 endpoint.
-
-In case the client contacts the tracker on an IPv6 interface, it may add
-an ``&ipv4=`` parameter with its IPv4 address or endpoint. The value MUST
-be either an IPv4 endpoint (address and port) or just an IPv4 address. If only
-an address is supplied, the port is assumed to be the same as the ``&port=``
-parameter.
+Announcing
+==========
 
-The endpoints are encoded as strings as defined by `RFC 2732`_.
+For each IP address the client can receive incoming peer connections over, it
+SHOULD announce to the tracker with that as the source IP address.
```

Comment:
*with that as the source* address is a bit vague due to the indirect object and source address binding being a lesser-known feature of sockets. It should be made explicit that the listen address must be used when binding the source of the http connection if the client intends to announce a specific address instead of any random one that the operating system chooses.

Reply:
yeah, I also realized that the restriction of a compatible address family is not sufficient. You also need to be able to route. e.g. if you accept incoming connections over loopback, it's non-sensical to announce with that as the source IP.

I will improve this.

Reply:
I think this is addressed now